src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -19,13 +20,32 @@ class TableStatistics;
 AbstractLQPNode::AbstractLQPNode(LQPNodeType node_type) : _type(node_type) {}
 
 std::shared_ptr<AbstractLQPNode> AbstractLQPNode::deep_copy() const {
+  auto copied_left_child = left_child() ? left_child()->deep_copy() : std::shared_ptr<AbstractLQPNode>();
+  auto copied_right_child = this->right_child() ? this->right_child()->deep_copy() : std::shared_ptr<AbstractLQPNode>();
+
   // We cannot use the copy constructor here, because it does not work with shared_from_this()
-  auto deep_copy = _deep_copy_impl();
-  if (_children[0]) deep_copy->set_left_child(_children[0]->deep_copy());
-  if (_children[1]) deep_copy->set_right_child(_children[1]->deep_copy());
+  auto deep_copy = _deep_copy_impl(copied_left_child, copied_right_child);
+  deep_copy->set_left_child(copied_left_child);
+  deep_copy->set_right_child(copied_right_child);
+
   return deep_copy;
 }
 
+LQPColumnReference AbstractLQPNode::adapt_column_reference_to_different_lqp(
+    const LQPColumnReference& column_reference, const std::shared_ptr<AbstractLQPNode>& original_lqp,
+    const std::shared_ptr<AbstractLQPNode>& copied_lqp) {
+  Assert(original_lqp->output_column_count() == copied_lqp->output_column_count(), "lqp_copy must be a copy of this");
```

Comment:
This method is static, so "this" does not exist.

Reply:
done